src/query/procedure/mg_procedure_impl.cpp

```diff
@@ -2679,6 +2679,128 @@ mgp_error mgp_graph_create_edge(mgp_graph *graph, mgp_vertex *from, mgp_vertex *
       result);
 }
 
+mgp_error mgp_graph_change_edge_from(struct mgp_graph *graph, struct mgp_edge *e, struct mgp_vertex *new_from,
+                                     mgp_memory *memory, mgp_edge **result) {
+  return WrapExceptions(
+      [&]() -> mgp_edge * {
+        auto *ctx = graph->ctx;
+#ifdef MG_ENTERPRISE
+        if (memgraph::license::global_license_checker.IsEnterpriseValidFast() && ctx && ctx->auth_checker &&
+            !ctx->auth_checker->Has(e->impl, memgraph::query::AuthQuery::FineGrainedPrivilege::CREATE_DELETE)) {
+          throw AuthorizationException{"Insufficient permissions for changing an edge!"};
+        }
+#endif
+        if (!MgpGraphIsMutable(*graph)) {
+          throw ImmutableObjectException{"Cannot remove an edge from an immutable graph!"};
+        }
+
+        auto edge = std::visit(memgraph::utils::Overloaded{
+                                   [&](memgraph::query::DbAccessor *accessor) {
+                                     return accessor->ChangeEdgeFrom(
+                                         &e->impl, &std::get<memgraph::query::VertexAccessor>(new_from->impl));
+                                   },
+                                   [&](memgraph::query::SubgraphDbAccessor *accessor) {
+                                     return accessor->ChangeEdgeFrom(
+                                         &e->impl, &std::get<memgraph::query::SubgraphVertexAccessor>(new_from->impl));
+                                   }},
+                               graph->impl);
+
+        if (edge.HasError()) {
+          switch (edge.GetError()) {
+            case memgraph::storage::Error::NONEXISTENT_OBJECT:
+              LOG_FATAL("Query modules shouldn't have access to nonexistent objects when changing an edge!");
+            case memgraph::storage::Error::DELETED_OBJECT:
+              LOG_FATAL("The edge was already deleted.");
+            case memgraph::storage::Error::PROPERTIES_DISABLED:
+            case memgraph::storage::Error::VERTEX_HAS_EDGES:
+              LOG_FATAL("Unexpected error when removing an edge.");
+            case memgraph::storage::Error::SERIALIZATION_ERROR:
+              throw SerializationException{"Cannot serialize changing an edge."};
+          }
+        }
+
+        if (ctx->trigger_context_collector) {
+          ctx->trigger_context_collector->RegisterDeletedObject(e->impl);
+          ctx->trigger_context_collector->RegisterCreatedObject(*edge);
+        }
+
+        return std::visit(
+            memgraph::utils::Overloaded{
+                [memory, edge, new_from](memgraph::query::DbAccessor *) {
```

Comment:
It makes sense to add trailing return type so we are sure everything is okay
```suggestion
                [&memory, &edge, &new_from](memgraph::query::DbAccessor *) -> mgp_edge* {
```